From 34254e42101b304ab5e232dbc9294ced3084e6ec Mon Sep 17 00:00:00 2001
From: Cheyenne Wills <cwills@sinenomine.net>
Date: Fri, 9 May 2025 15:06:51 -0600
Subject: [PATCH 08/10] cf: check for dentry flag macros/enums

Linux 6.15 commit:
    dcache: convert dentry flag macros to enum (b2b4483b5d05)
converted the dentry flag macros to an enum.

A preprocessor #ifdef can no longer be used to test to see if the symbol
is present.

Use AC_CHECK_LINUX_SYMBOL to test for the DCACHE_NEED_AUTOMOUNT and
DCACHE_NFSFS_RENAMED flags.

Change-Id: Id03e6434e47df8bfb52c786ddd25ff1dac56ba63
---
 src/afs/LINUX/osi_compat.h    | 4 ++--
 src/cf/linux-kernel-symbol.m4 | 2 ++
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/afs/LINUX/osi_compat.h b/src/afs/LINUX/osi_compat.h
index e8cc9b357..b26b9fd7f 100644
--- a/src/afs/LINUX/osi_compat.h
+++ b/src/afs/LINUX/osi_compat.h
@@ -48,7 +48,7 @@ SetPageError(struct page *p)
 # define kthread_complete_and_exit complete_and_exit
 #endif
 
-#if defined(STRUCT_DENTRY_OPERATIONS_HAS_D_AUTOMOUNT) && !defined(DCACHE_NEED_AUTOMOUNT)
+#if defined(STRUCT_DENTRY_OPERATIONS_HAS_D_AUTOMOUNT) && !defined(HAVE_LINUX_SYMBOL_DCACHE_NEED_AUTOMOUNT)
 # define DCACHE_NEED_AUTOMOUNT DMANAGED_AUTOMOUNT
 #endif
 
@@ -123,7 +123,7 @@ afs_posix_test_lock(struct file *fp, struct file_lock *flp) {
 #endif
 }
 
-#ifdef DCACHE_NFSFS_RENAMED
+#ifdef HAVE_LINUX_SYMBOL_DCACHE_NFSFS_RENAMED
 static inline void
 afs_linux_clear_nfsfs_renamed(struct dentry *dp) {
     spin_lock(&dp->d_lock);
diff --git a/src/cf/linux-kernel-symbol.m4 b/src/cf/linux-kernel-symbol.m4
index 5400a736a..ae12a5272 100644
--- a/src/cf/linux-kernel-symbol.m4
+++ b/src/cf/linux-kernel-symbol.m4
@@ -1,3 +1,5 @@
 AC_DEFUN([OPENAFS_LINUX_KERNEL_SYMBOL_CHECKS],[
 dnl Type existence checks
+AC_CHECK_LINUX_SYMBOL([DCACHE_NEED_AUTOMOUNT], [dcache.h])
+AC_CHECK_LINUX_SYMBOL([DCACHE_NFSFS_RENAMED], [dcache.h])
 ])
-- 
2.49.0

